src/WpfMath/TexFormulaParser.cs

```diff
@@ -412,11 +561,209 @@ private static SourceSpan ReadElement(SourceSpan value, ref int position)
 
                         throw new TexParseException($"Color {colorName} not found");
                     }
+
+                case "vmatrix":
+                    {
+                        if (position == value.Length)
+                            throw new TexParseException("illegal end!");
+                        SkipWhiteSpace(value, ref position);
+
+                        var matrixsource = ReadElement(value, ref position);
+
+                        var cells = GetMatrixData(formula, matrixsource);
+                        return new VmatrixAtom(matrixsource, new MatrixAtom(matrixsource, cells));
+                    }
+
+                case "Vmatrix":
+                    {
+                        if (position == value.Length)
+                            throw new TexParseException("illegal end!");
+                        SkipWhiteSpace(value, ref position);
+
+                        var matrixsource = ReadElement(value, ref position);
+
+                        var cells = GetMatrixData(formula, matrixsource);
+                        return new VVMatrixAtom(matrixsource, new MatrixAtom(matrixsource, cells));
+                    }
+
             }
 
             throw new TexParseException("Invalid command.");
         }
 
+        /// <summary>
+        /// Retrives the cells of a matrix from a given source.
+        /// </summary>
+        /// <param name="matrixsource">The source to retrieve the 2D-Array of atoms from.</param>
+        private List<List<Atom>> GetMatrixData(TexFormula formula, SourceSpan matrixsource, bool allowEmptyCells = true, bool parseasArrays = false)
+        {
+            List<List<StringBuilder>> rowdata = new List<List<StringBuilder>>() { new List<StringBuilder>() { new StringBuilder() } };
+            int rows = 0;
+            int cols = 0;
+            //how many characters the next row should skip for its sourcespan to start
+            int rowindent = 0;
```

Comment:
This variable most of the time holds invalid values, beause it gets assigned on the parsing stage, but gets used much later, when it has some last value from the matrix.